sdk/core/Azure.Core/src/AzureKeyCredential.cs

```diff
@@ -13,16 +12,16 @@ namespace Azure
     /// </summary>
     public class AzureKeyCredential
     {
-        private string _key;
+        private readonly KeyCredential _credential;
```

Comment:
I like the idea of sealing ClientModel credential types to preserve atomicity of reads - the result is we need to use delegation rather than inheritance in Azure.Core credential types.

Reply:
Why do we need to seal to do atomic reads?

Reply:
If a type derives from this type, it could add a property that needs to be read atomically with key

Reply:
I think sealing is too big of a hammer for this problem. We can simply not add fields to subtypes. If we use delegation AzureKeyCredential cannot be passed to APIs taking KeyCredential, that I think we should have as an option.

Reply:
+1 to not `seal`ing.  I understand the concern at a high level, but I can't really imagine anyone deriving from one of our credentials and expecting it to impact the auth story used by our pipeline policies.

Reply:
Sounds good - I will back that out before merging.